ci/management/commands/generate_reports.py

```diff
@@ -303,3 +352,68 @@ def handle_health_worker_registrations(
                 'Cadre': operator_details.get('role'),
                 'Number of Registrations': count,
             })
+
+    def handle_enrollments(self, sheet, sbm_client, ids_client, start_date,
+                           end_date):
+
+        sheet.set_header([
+            'Message set',
+            'Roleplayer',
+            'Currently enrolled (snapshot at point y)',
+            'Cumulatively enrolled (in last period x)',
+            'Of the cumulatively enrolled (in last period x), number which '
+            'opted out',
+            'Of the cumulatively enrolled (in last period x), the number which'
+            ' completed messages',
+        ])
+
+        subscriptions = self.get_subscriptions(
+            sbm_client,
+            created_before=end_date.isoformat())
+
+        data = {}
+        for idx, subscription in enumerate(subscriptions):
+            messageset = self.get_messageset(
+                            sbm_client, subscription['messageset'])
+            identity = self.get_identity(ids_client, subscription['identity'])
+
+            messageset_name = messageset['short_name'].split('.')[0]
+
+            receiver_role = 'None'
+            if identity:
+                receiver_role = identity.get('details', {}).get(
+                                    'receiver_role', 'None')
+
+            key = '%s_%s' % (messageset_name, receiver_role)
```

Comment:
We can use tuples for dictionary keys, so maybe it's better to do `data[(messageset_name, receiver_role)] = ...`. That way we can also get them out when iterating, and don't need to store them again in the dictionary. ie.
```
for (message_name, receiver_role), d in data.items():
    ...
```